### PR TITLE
EDRS policy configuration for SDDC resource

### PR DESCRIPTION
**Pull request by saloneerege**
1.  Updated resourceSddcRead method to store information about Primary cluster in cluster_info 
2. Updated resourceSddcRead method to store current Default EDRS policy for primary cluster.
3. Added code in resourceSddcUpdate to change the EDRS policy for primary cluster.

### CODE REVIEW

**Comment by saloneerege** [comment]
This code has not been tested yet.

**Review by sumitAgrawal007 — COMMENTED** [review]
LGTM

File: vendor/modules.txt [hunk 1]
```diff
@@ -262,16 +262,19 @@ github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client/metadata
...
 github.com/vmware/vsphere-automation-sdk-go/services/vmc
 github.com/vmware/vsphere-automation-sdk-go/services/vmc/model
 github.com/vmware/vsphere-automation-sdk-go/services/vmc/orgs
 github.com/vmware/vsphere-automation-sdk-go/services/vmc/orgs/account_link
 github.com/vmware/vsphere-automation-sdk-go/services/vmc/orgs/sddcs
-# github.com/vmware/vsphere-automation-sdk-go/services/vmc/draas v0.2.0
+# github.com/vmware/vsphere-automation-sdk-go/services/vmc/autoscaler v0.1.0
+github.com/vmware/vsphere-automation-sdk-go/services/vmc/autoscaler/api/orgs/sddcs/clusters
+github.com/vmware/vsphere-automation-sdk-go/services/vmc/autoscaler/model
+# github.com/vmware/vsphere-automation-sdk-go/services/vmc/draas v0.3.0
```
Comment (shylasrinivas) [inline vendor/modules.txt:291]: why do we have these commented imports?

File: vmc/resource_vmc_sddc.go [hunk 2]
```diff
@@ -343,16 +373,26 @@ func resourceSddcRead(d *schema.ResourceData, m interface{}) error {
...
+
+	d.Set("cluster_id", primaryCluster.ClusterId)
+	cluster := map[string]string{}
+	cluster["cluster_name"] = *primaryCluster.ClusterName
+	cluster["cluster_state"] = *primaryCluster.ClusterState
+	cluster["host_instance_type"] = *primaryCluster.EsxHostInfo.InstanceType
+	d.Set("cluster_info", cluster)
+
+	edrsPolicyClient := autoscalercluster.NewDefaultEdrsPolicyClient(connector)
+	edrsPolicy, err := edrsPolicyClient.Get(orgID, sddcID, primaryCluster.ClusterId)
```
Comment (shylasrinivas) [inline vmc/resource_vmc_sddc.go:401]: not checking for err here?

File: vmc/resource_vmc_sddc.go [hunk 3]
```diff
@@ -441,6 +481,28 @@ func resourceSddcUpdate(d *schema.ResourceData, m interface{}) error {
...
+			MinHosts:   &minHosts,
+			MaxHosts:   &maxHosts,
+		}
+		_, err := edrsPolicyClient.Post(orgID, sddcID, clusterID, *edrsPolicy)
+		if err != nil {
+			return HandleUpdateError("EDRS Policy", err)
+		}
+
+		// To be removed once the API spec has been fixed to return the task ID
+		time.Sleep(2 * time.Minute)
```
Comment (shylasrinivas) [inline vmc/resource_vmc_sddc.go]: 2 min wait feels long..

File: vendor/modules.txt (hunk 1 above)
Comment (saloneerege) [reply vendor/modules.txt:291]: I have not manually modified this file. I only ran go mod vendor command

File: vmc/resource_vmc_sddc.go (hunk 2 above)
Comment (saloneerege) [reply vmc/resource_vmc_sddc.go:401]: Good catch. I will add it

File: vmc/resource_vmc_sddc.go (hunk 3 above)
Comment (saloneerege) [reply vmc/resource_vmc_sddc.go]: This 2 wait is going to be removed before the PR gets merged to the master branch. The OpenAPI spec for EDRS was missing the task ID. This is a temporary workaround. Once the issue has been fixed we will be removing this wait.

**Review by shylasrinivas — APPROVED** [review]

File: vmc/resource_vmc_sddc.go (hunk 3 above)
Comment (saloneerege) [reply vmc/resource_vmc_sddc.go]: Done

**Pull request closed without merging**